src/ducks/Watchlists/Widgets/TopBar/BaseActions/index.js

```diff
@@ -0,0 +1,134 @@
+} from './Items'
+import styles from './index.module.scss'
+
+const Actions = ({
+  watchlist,
+  type,
+  onClick,
+  isAuthor,
+  isAuthorLoading,
+  refetchAssets
+}) => {
+  const [lists] = useUserWatchlists(type)
+  const [updateWatchlist, { loading }] = useUpdateWatchlist(type)
+  const [isMenuOpened, setIsMenuOpened] = useState(false)
+  const [showPanel, setShowPanel] = useState(true)
+
+  useEffect(() => {
+    const panelVisibilityChange = ({ detail }) => {
+      const isShowPanel = detail === 'show'
+      if (!isShowPanel) setShowPanel(isShowPanel)
+      setIsMenuOpened(!isShowPanel)
+    }
+    window.addEventListener(
+      'panelVisibilityChange',
+      panelVisibilityChange,
+      false
+    )
+    return () =>
+      window.removeEventListener(
+        'panelVisibilityChange',
+        panelVisibilityChange,
+        false
+      )
+  }, [])
+
+  if (!watchlist.id || isAuthorLoading) {
+    return null
+  }
+
+  if (!isAuthor) {
+    return (
+      <div onClick={onClick} className={styles.container}>
+        <SaveAsAction
+          type={type}
+          watchlist={watchlist}
+          trigger={<NonAuthorTrigger />}
+        />
+      </div>
+    )
+  }
+
+  const { id, name } = watchlist
+  const title = getTitleByWatchlistType(type)
+  const showDelete = isAuthor && (type !== SCREENER || lists.length > 1)
+
+  const onEditApprove = props =>
+    updateWatchlist(watchlist, { ...props }).then(() => {
+      setIsMenuOpened(false)
+      notifyUpdate(title)
+      refetchAssets && refetchAssets()
+    })
+
+  return (
+    <div onClick={onClick} className={styles.container}>
+      <ContextMenu
+        trigger={
+          <Trigger
+            type={type}
+            title={title}
+            isLoading={loading}
+            watchlist={watchlist}
+            onPrimaryAction={onEditApprove}
+            openMenu={() => {
+              setShowPanel(true)
+              setIsMenuOpened(true)
+            }}
+          />
+        }
+        align='start'
+        position='bottom'
+        open={isMenuOpened}
+        passOpenStateAs='isActive'
+        onClose={() => setIsMenuOpened(false)}
+      >
+        <Panel
+          variant='modal'
+          className={showPanel ? styles.wrapper : styles.hidePanel}
+        >
+          <Edit
+            type={type}
+            title={title}
+            watchlist={watchlist}
+            isLoading={loading}
+            onSubmit={onEditApprove}
+          />
+          <SaveAs type={type} watchlist={watchlist} />
+          <New type={type} />
+          {type === 'PROJECT' && <Copy watchlist={watchlist} />}
+          {showDelete && <Delete id={id} name={name} title={title} />}
+        </Panel>
+      </ContextMenu>
+    </div>
+  )
+}
+
+Actions.propTypes = {
+  onClick: PropTypes.func,
+  isAuthor: PropTypes.bool.isRequired,
+  isAuthorLoading: PropTypes.bool.isRequired
+}
+
+Actions.defaultProps = {
+  watchlist: {},
+  onClick: _ => _
+}
+
+export default Actions
```

Comment:
Copy of the `src/ducks/Watchlists/Widgets/TopPanel/BaseActions/index.js` module. Reuse existing components. If they can't be used directly - create an abstraction for a general use.